### PR TITLE
fix[experiments]: make rn_experiments a separate entrypoint

### PR DESCRIPTION
**Pull request by hoxyq**
# Summary

We've recently landed https://github.com/facebookexperimental/rn-chrome-devtools-frontend/pull/44, which encapsulated RN-related experiments logic. This implementation doesn't work in release build, because `rn_experiments.js` file is not a separate entrypoint. For release builds, there is no separate module for this script, it gets copied and inlined into different entrypoints: `main`, `rn_fusebox`, `rn_inspector`, which means that there is no module-level preserved state and `setIsReactNativeEntryPoint` function won't work. Because of this, `isReactNativeEntryPoint` is always evaluated to `false` and these experiments are marked as non-configurable => not being listed on settings page.

- Also ran `npm run check-lint-js` to fix a bunch of linting errors across project.

# Test plan

- Manual code inspection
- Opened generated release build and validated that RN experiments are listed
Screenshot 2024-04-16 at 15 35 56

# Upstreaming plan

- [ ] This commit should be sent as a patch to the upstream `devtools-frontend` repo. I've reviewed the [contribution guide](https://docs.google.com/document/d/1WNF-KqRSzPLUUfZqQG5AFeU_Ll8TfWYcJasa_XGf7ro/edit#heading=h.9kj7femz1xg5).
- [x] This commit is React Native-specific and cannot be upstreamed.

### CODE REVIEW

**Review by motiz88 — APPROVED** [review]
Broadly lgtm, just some naming thoughts

File: front_end/core/rn_experiments/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "rn_experiments.ts"
+  ]
+
+  deps = [
+    "../../core/root:bundle",
+  ]
+}
+
+devtools_entrypoint("bundle") {
+  entrypoint = "bundle.ts"
```
[the remark this answers is not in the document]
Comment (hoxyq) [reply front_end/core/rn_experiments/BUILD.gn:20]: Renamed `bundle` to `rn_experiments`, looks like there is a convention across repo to name entrypoint file same as the folder name

Also renamed `rn_experiments` to `experimentsImpl`.

File: front_end/entrypoints/main/MainImpl.ts [hunk 2]
```diff
@@ -429,7 +429,7 @@ export class MainImpl {
 
     // React Native experiments need to be registered for all entry points so
     // that they can be checked everywhere.
-    RNExperiments.copyInto(Root.Runtime.experiments, '[React Native] ');
+    RNExperiments.RNExperiments.copyInto(Root.Runtime.experiments, '[React Native] ');
```
[the remark this answers is not in the document]
Comment (hoxyq) [reply front_end/entrypoints/main/MainImpl.ts:432]: Renamed, now its `RNExperiments. RNExperimentsImpl.Instance.copyInto(...)`

File: front_end/core/rn_experiments/rn_experiments.ts [hunk 3]
```diff
@@ -46,10 +45,10 @@ const state = {
  * Set whether the current entry point is a React Native entry point.
  * This must be called before constructing MainImpl.
  */
-export function setIsReactNativeEntryPoint(value: boolean) {
+export function setIsReactNativeEntryPoint(value: boolean): void {
   if (state.didInitializeExperiments) {
     throw new Error(
-      'setIsReactNativeEntryPoint must be called before constructing MainImpl'
+      'setIsReactNativeEntryPoint must be called before constructing MainImpl',
```
[the remark this answers is not in the document]
Comment (hoxyq) [reply front_end/core/rn_experiments/rn_experiments.ts:51]: Yes, running this exits the process with 0-code

**Pull request merged**